File: examples/java/Quickstart.java
import ai.cogneris.documentai.CognerisClient;
import ai.cogneris.documentai.model.DocumentJobSubmitOperation;
import java.nio.file.Path;
import java.time.Duration;
import java.util.Objects;
import java.util.UUID;

public final class Quickstart {
    private static final String USAGE = """
            Usage:
              java Quickstart extract <file>
              java Quickstart submit <input-reference>
              java Quickstart get <job-id>
              java Quickstart wait <job-id>
              java Quickstart cancel <job-id>
            """;

    private Quickstart() { }

    public static void main(String[] args) throws InterruptedException {
        if (args.length != 2) throw new IllegalArgumentException(USAGE);
        String apiKey = Objects.requireNonNull(
                System.getenv("COGNERIS_API_KEY"),
                "COGNERIS_API_KEY is required");
        var region = switch (System.getenv().getOrDefault("COGNERIS_REGION", "us")) {
            case "us" -> CognerisClient.Region.US;
            case "eu" -> CognerisClient.Region.EU;
            default -> throw new IllegalArgumentException("COGNERIS_REGION must be either us or eu.");
        };
        var client = new CognerisClient(new CognerisClient.Options(apiKey, region, null));

        switch (args[0]) {
            case "extract" -> {
                var envelope = client.extract(Path.of(args[1]), null);
                System.out.println("extracted; hasErrors=" + envelope.getHasErrors());
            }
            case "submit" -> {
                var submission = client.submitJob(DocumentJobSubmitOperation.EXTRACTION, args[1]);
                System.out.println("submitted; jobId=" + submission.getJobId());
            }
            case "get" -> {
                var job = client.getJob(UUID.fromString(args[1]));
                System.out.println("jobId=" + job.getJobId() + "; status=" + job.getStatus());
            }
            case "wait" -> {
                var job = client.waitForJob(
                        UUID.fromString(args[1]),
                        20,
                        Duration.ofMinutes(2),
                        Duration.ofSeconds(1));
                System.out.println("jobId=" + job.getJobId() + "; status=" + job.getStatus());
            }
            case "cancel" -> {
                var cancellation = client.cancelJob(UUID.fromString(args[1]));
                System.out.println("jobId=" + cancellation.getJobId()
                        + "; cancellationRequested=" + cancellation.getCancellationRequested());
            }
            default -> throw new IllegalArgumentException(USAGE);
        }
    }
}

File: examples/dotnet/Quickstart.cs
using Cogneris.DocumentAI;
using Cogneris.DocumentAI.Model;

const string usage = """
Usage:
  dotnet run -- extract <file>
  dotnet run -- submit <input-reference>
  dotnet run -- get <job-id>
  dotnet run -- wait <job-id>
  dotnet run -- cancel <job-id>
""";

var apiKey = Environment.GetEnvironmentVariable("COGNERIS_API_KEY");
if (string.IsNullOrWhiteSpace(apiKey))
    throw new InvalidOperationException("COGNERIS_API_KEY is required.");

var region = (Environment.GetEnvironmentVariable("COGNERIS_REGION") ?? "us") switch
{
    "us" => CognerisRegion.Us,
    "eu" => CognerisRegion.Eu,
    _ => throw new ArgumentException("COGNERIS_REGION must be either us or eu.")
};

if (args.Length != 2)
    throw new ArgumentException(usage);

await using var client = new CognerisClient(new CognerisClientOptions(apiKey, region));

switch (args[0])
{
    case "extract":
        await using (var input = File.OpenRead(args[1]))
        {
            var envelope = await client.ExtractAsync(input, Path.GetFileName(args[1]));
            Console.WriteLine($"extracted; hasErrors={envelope.HasErrors}");
        }
        break;
    case "submit":
        var submission = await client.SubmitJobAsync(
            DocumentJobSubmitOperation.Extraction,
            args[1]);
        Console.WriteLine($"submitted; jobId={submission.JobId}");
        break;
    case "get":
        var current = await client.GetJobAsync(Guid.Parse(args[1]));
        Console.WriteLine($"jobId={current.JobId}; status={current.Status}");
        break;
    case "wait":
        var completed = await client.WaitForJobAsync(Guid.Parse(args[1]));
        Console.WriteLine($"jobId={completed.JobId}; status={completed.Status}");
        break;
    case "cancel":
        var cancellation = await client.CancelJobAsync(Guid.Parse(args[1]));
        Console.WriteLine($"jobId={cancellation.JobId}; cancellationRequested={cancellation.CancellationRequested}");
        break;
    default:
        throw new ArgumentException(usage);
}
